src/lib/tokenizer.js

```diff
@@ -0,0 +1,88 @@
+class Tokenizer {
+  static theTokenizer;
+
+  constructor(content, literalsList) {
+    this.program = content;
+    this.literals = literalsList;
+    this.tokens = [];
+    this.currentToken = 0;
+
+    this.tokenize();
+  }
+
+  tokenize() {
+    let tokenizedProgram = this.program;
+    tokenizedProgram = tokenizedProgram.replace(/\n/g, '');
+    console.log(this.program);
+
+    this.literals.forEach(
+        s => {
+          tokenizedProgram = tokenizedProgram.replace(s, `_${s}_`);
+          console.log(tokenizedProgram);
+        });
+    tokenizedProgram = tokenizedProgram.replace(/__/g, '_');
+    console.log(tokenizedProgram);
+    const temparray = tokenizedProgram.split('_');
+    this.tokens = temparray.slice(1);
+    console.log(this.tokens);
+  }
+
+  checkNext() {
+    let token = '';
+
+    if (this.currentToken < this.tokens.length) {
+      token = this.tokens[this.currentToken];
+    } else {
+      token = 'NO_MORE_TOKENS';
+    }
+
+    return token;
+  }
+
+  getNext() {
+    let token = '';
+
+    if (this.currentToken < this.tokens.length) {
+      token = this.tokens[this.currentToken];
+      this.currentToken++;
+    } else {
+      token = 'NULLTOKEN';
+    }
+
+    return token;
+  }
+
+  checkToken(regexp) {
+    const s = this.checkNext();
+    console.log(`comparing: |${s}|  to  |${regexp}|`);
+    return s.match(regexp);
+  }
+
+  getAndCheckToken(regexp) {
+    const s = this.getNext();
+
+    if (!s.match(regexp)) {
+      throw Error(
+          `Unexpected next token for Parsing! Expected something matching: ${regexp} but got: ${s}`);
+    }
+    console.log(`matched: ${s}  to  ${regexp}`);
+
+    return s;
+  }
```

Comment:
String::match returns an array of local matches rather than seeing if the entire string matches, I'm not experienced with regex, is the best way to fix this to encase the regexp like so? `` `^{regexp}$` ``

Reply:
I think it's the user's responsibility to pass a regexp object to the `getAndCheckToken()` method. For example, if the user wants to know if the next token starts with `Title`, he should pass the regexp object `/^Title/` to the method.